source/utils/schedule.py

```diff
@@ -0,0 +1,189 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2024 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+from datetime import datetime
+from enum import Enum, auto
+import threading
+import time
+from typing import Callable
+
+import schedule
+
+from logHandler import log
+import NVDAState
+
+scheduleThread: "ScheduleThread | None" = None
+
+
+class ThreadTarget(Enum):
+	GUI = auto()
+	"""
+	Uses wx.CallAfter to run the job on the GUI thread.
+	This is encouraged for tasks that interact with the GUI, such as dialogs.
+	"""
+
+	DAEMON = auto()
+	"""
+	Uses threading.Thread(daemon=True) to run the job in the background.
+	"""
+
+	CUSTOM = auto()
+	"""
+	No thread target.
+	Runs directly and blocks `scheduleThread`.
+	Run code on a custom thread to ensure `scheduleThread` is not blocked.

```

Comment:
Here, for me it's not clear if custom thread means a thread created using a class derived from the ScheduleThread class, without using the scheduleThread global variable, or if this means to use the ThreadTarget.CUSTOM with the scheduleThread global variable.

Reply:
I've added better docs now, does this help?